### PR TITLE
PoC new SDK

### PR DESCRIPTION
**Pull request by roneli**

### CODE REVIEW

File: resources/ec2_instances.go [hunk 1]
```diff
@@ -351,7 +352,7 @@ func Ec2Instances() *schema.Table {
 				Resolver:    fetchEc2InstanceBlockDeviceMappings,
 				Columns: []schema.Column{
 					{
-						Name:        "instance_id",
+						Name:        "instance_cq_id",
```
Comment (fdistorted) [inline resources/ec2_instances.go:355]: Should we add instance_id field that points to the instance id primary key of the parent?

Comment (roneli) [reply resources/ec2_instances.go:363]: yes, we can also do that with ParentFieldResolver

File: resources/iam_policies.go [hunk 2]
```diff
@@ -89,11 +92,17 @@ func IamPolicies() *schema.Table {
 				Resolver:    fetchIamPolicyVersions,
 				Columns: []schema.Column{
```
Comment (fdistorted) [inline resources/iam_policies.go:93]: Options are missing

File: resources/lambda_functions.go [hunk 3]
```diff
@@ -678,6 +703,7 @@ func LambdaFunctions() *schema.Table {
 						Name:        "aws_lambda_function_version_layers",
 						Description: "An AWS Lambda layer (https://docs.aws.amazon.com/lambda/latest/dg/configuration-layers.html). ",
 						Resolver:    fetchLambdaFunctionVersionLayers,
+						Options:     schema.TableCreationOptions{PrimaryKeys: []string{"function_version_id", "arn"}},
 						Columns: []schema.Column{
 							{
 								Name:        "function_version_id",
```
Comment (fdistorted) [inline resources/lambda_functions.go:709]: should be function_version_cq_id

File: resources/s3_buckets.go [hunk 4]
```diff
@@ -21,6 +21,7 @@ func S3Buckets() *schema.Table {
 		IgnoreError:          client.IgnoreAccessDeniedServiceDisabled,
 		DeleteFilter:         client.DeleteAccountFilter,
 		PostResourceResolver: resolveS3BucketsAttributes,
+		Options:              schema.TableCreationOptions{PrimaryKeys: []string{"account_id", "region", "name"}},
```
Comment (fdistorted) [inline resources/s3_buckets.go:24]: As I know buckets have global name across all the regions

File: resources/waf_rules.go [hunk 5]
```diff
@@ -75,7 +78,7 @@ func WafRules() *schema.Table {
 					},
```
Comment (fdistorted) [inline resources/waf_rules.go:78]: maybe  `data_id` should be  renamed  to `id`.

**Review by fdistorted — COMMENTED** [review]
I found a few things to look at.
I think we need to set some rules to make naming more consistent and to have consistent priority of columns that can be used as pk

File: resources/cloudwatchlogs_filters.go [hunk 6]
```diff
@@ -18,6 +18,7 @@ func CloudwatchlogsFilters() *schema.Table {
 		Multiplex:    client.AccountRegionMultiplex,
 		IgnoreError:  client.IgnoreAccessDeniedServiceDisabled,
 		DeleteFilter: client.DeleteAccountRegionFilter,
+		Options:      schema.TableCreationOptions{PrimaryKeys: []string{"account_id", "region", "name", "log_group_name"}},
```
[the remark this answers is not in the document]
Comment (roneli) [reply resources/cloudwatchlogs_filters.go:21]: Not required since rc5.

File: resources/waf_rules.go (hunk 5 above)
Comment (roneli) [reply resources/waf_rules.go:78]: In relations I opted to not change. We can discuss this further

**Pull request merged**